### ISSUE TITLE
Webpacker::Manifest::MissingEntryError in Pages#home

### ISSUE THREAD

**Issue by LeeRenJie** [rails/webpacker#2992]
FIX: Was able to get around it by recreating the site using Node v14.16.1. Previously I had been using v15.14.0.
![image](https://user-images.githubusercontent.com/67599130/115954497-fc48cd00-a523-11eb-825d-57c91bd9e78c.png)

I've been trying to solve this issue for days now. Looked through the solutions on the previous issues and stack overflow but it still did not work. 

I have provided some file's code below, please let me know if any specific code in other files are needed!

I downloaded Ubuntu 20.4 on the Linux subsystem on windows and using asdf as the package manager.

.tools-version:
```
ruby 2.7.3
nodejs 15.14.0
yarn 1.22.10
```

package,json file:
```js
{
  "name": "alpha-blog",
  "private": true,
  "dependencies": {
    "@rails/actioncable": "^6.0.0",
    "@rails/activestorage": "^6.0.0",
    "@rails/ujs": "^6.0.0",
    "@rails/webpacker": "^6.0.0-beta.7",
    "node-sass": "^5.0.0",
    "turbolinks": "^5.2.0",
    "webpack": "^4",
    "webpack-cli": "^4.6.0"
  },
  "version": "0.1.0",
  "devDependencies": {
    "webpack-dev-server": "^3.11.2"
  }
}
```

webpacker.yml:
```yml
# Note: You must restart bin/webpack-dev-server for changes to take effect

default: &default
  source_path: app/javascript
  source_entry_path: packs
  public_root_path: public
  public_output_path: packs
  cache_path: tmp/cache/webpacker
  webpack_compile_output: true

  # Additional paths webpack should lookup modules
  # ['app/assets', 'engine/foo/app/assets']
  additional_paths: []

  # Reload manifest.json on all requests so we reload latest compiled packs
  cache_manifest: false

  # Extract and emit a css file
  extract_css: false

  static_assets_extensions:
    - .jpg
    - .jpeg
    - .png
    - .gif
    - .tiff
    - .ico
    - .svg
    - .eot
    - .otf
    - .ttf
    - .woff
    - .woff2

  extensions:
    - .mjs
    - .js
    - .sass
    - .scss
    - .css
    - .module.sass
    - .module.scss
    - .module.css
    - .png
    - .svg
    - .gif
    - .jpeg
    - .jpg

development:
  <<: *default
  compile: true

  # Reference: https://webpack.js.org/configuration/dev-server/
  dev_server:
    https: false
    host: localhost
    port: 3035
    public: localhost:3035
    hmr: false
    # Inline should be set to true if using HMR
    inline: true
    overlay: true
    compress: true
    disable_host_check: true
    use_local_ip: false
    quiet: false
    pretty: false
    headers:
      'Access-Control-Allow-Origin': '*'
    watch_options:
      ignored: '**/node_modules/**'


test:
  <<: *default
  compile: true

  # Compile test packs to a separate directory
  public_output_path: packs-test

production:
  <<: *default

  # Production depends on precompilation of packs prior to booting for performance.
  compile: false

  # Extract and emit a css file
  extract_css: true

  # Cache manifest.json for performance
  cache_manifest: true
```

application.html.erb file
```erb
<!DOCTYPE html>
<html>
  <head>
    <title>AlphaBlog</title>
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <%= csrf_meta_tags %>
    <%= csp_meta_tag %>

    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
  </head>

  <body>
    <%= yield %>
  </body>
</html>
```

**Comment by moritzstuker** [comment]
I have the exact same issue.

My guess is that the problem is linked with the fact that among the dependencies there's node-sass, which has been deprecated.

I say this, because when I run `bundle exec rails webpacker:install`, all hell breaks loose after the installer deals with node-sass...

````
User@MacBook-Air Rails % bundle exec rails webpacker:install
      create  config/webpacker.yml
Copying webpack core config
      create  config/webpack
      create  config/webpack/development.js
      create  config/webpack/environment.js
      create  config/webpack/production.js
      create  config/webpack/test.js
Copying postcss.config.js to app root directory
      create  postcss.config.js
Copying babel.config.js to app root directory
      create  babel.config.js
Copying .browserslistrc to app root directory
      create  .browserslistrc
Creating JavaScript app source directory
      create  app/javascript
      create  app/javascript/packs/application.js
       apply  /Users/User/.rvm/gems/ruby-2.6.6/gems/webpacker-4.3.0/lib/install/binstubs.rb
  Copying binstubs
       exist    bin
      create    bin/webpack
      create    bin/webpack-dev-server
      append  .gitignore
Installing all JavaScript dependencies [4.3.0]
         run  yarn add @rails/webpacker@4.3.0 from "."
yarn add v1.22.10
[1/4] 🔍  Resolving packages...
warning @rails/webpacker > node-sass > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @rails/webpacker > node-sass > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @rails/webpacker > node-sass > request > har-validator@5.1.5: this library is no longer supported
warning @rails/webpacker > webpack > watchpack > watchpack-chokidar2 > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning @rails/webpacker > webpack > watchpack > watchpack-chokidar2 > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning @rails/webpacker > webpack > micromatch > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
warning @rails/webpacker > webpack > micromatch > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
[-/3] ⡀ waiting...
[-/3] ⡀ waiting...
error /Users/User/Projects/Rails/node_modules/node-sass: Command failed.
Exit code: 1
Command: node scripts/build.js
Arguments: 
Directory: /Users/User/Projects/Rails/node_modules/node-sass
Output:
Building: /opt/homebrew/Cellar/node/16.0.0/bin/node /Users/User/Projects/Rails/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
````
[...]

**Comment by LeeRenJie** [comment]
@moritzstuker yep same goes for me.

**Comment by acouprie** [comment]
This issue is likely linked with mine [issue#2989](https://github.com/rails/webpacker/issues/2989), however I don't use node-sass but only  `"sass": "^1.32.10"` and `"sass-loader": "^11.0.1"`.

**Comment by jessecai33** [comment]
same issue here. Webpacker can't find related .js file in manifest.json.

**Comment by moritzstuker** [comment]
Though not a fix, here's a workaround: Install an alternative terminal such as [Hyper](https://hyper.is) and rerun all your commands with it. Why does it work? Because chances are you're encountering this error while on an ARM-based Apple computer, with which not all software is yet perfectly compatible. Hyper is however still (only) coded for x86 processors, which means that all commands run through is will also be executed as x86 (and translated with Rosetta2). Thus, it will suddenly magically work.

Of course, you could also choose to run your Terminal using Rosetta, as described [here](https://dev.to/courier/tips-and-tricks-to-setup-your-apple-m1-for-development-547g), though I found the solution of using a different Terminal more elegant.

**Comment by LeeRenJie** [comment]
> This issue is likely linked with mine [issue#2989](https://github.com/rails/webpacker/issues/2989), however I don't use node-sass but only `"sass": "^1.32.10"` and `"sass-loader": "^11.0.1"`.
> 
> You have no traces of the `manifest.json` on your project, right?

not sure what do you mean by this. I just started learning Ruby on Rails😅But if you are stating that there isn't a manifest.json file in my local then yes.

**Comment by LeeRenJie** [comment]
> Though not a fix, here's a workaround: Install an alternative terminal such as [Hyper](https://hyper.is) and rerun all your commands with it. Why does it work? Because chances are you're encountering this error while on an ARM-based Apple computer, with which not all software is yet perfectly compatible. Hyper is however still (only) coded for x86 processors, which means that all commands run through is will also be executed as x86 (and translated with Rosetta2). Thus, it will suddenly magically work.
> 
> Of course, you could also choose to run your Terminal using Rosetta, as described [here](https://dev.to/courier/tips-and-tricks-to-setup-your-apple-m1-for-development-547g), though I found the solution of using a different Terminal more elegant.

I have Hyper terminal but it is used for my windows OS. My project is currently in the Ubuntu OS which I downloaded through Windows Subsystem Linux. Therefore, I think I can only use the Ubuntu terminal and Powershell. Not sure about this. Thank you for the input tho😁

**Comment by samleibowitz** [comment]
I ran into this same issue while working through a tutorial yesterday and was able to get around it by recreating the site using Node v14.16.1. Previously I had been using v15.14.0.

**Comment by LeeRenJie** [comment]
> I ran into this same issue while working through a tutorial yesterday and was able to get around it by recreating the site using Node v14.16.1. Previously I had been using v15.14.0.

Thank you, finally solved my problem!

**Comment by jellycattime** [comment]
> I ran into this same issue while working through a tutorial yesterday and was able to get around it by recreating the site using Node v14.16.1. Previously I had been using v15.14.0.

Dude, I read hundreds of articles about this problem, tried everything possible, and my colleagues assured me that it could not be in the node. As a result, I found that somewhere in the project a version of 14x is needed, rolled back my node and it all worked. I spent half a week on this s***. Thank you, I would probably get fired hahahaha

**Comment by blefev** [comment]
Can confirm downgrading to node 14.16.1 was the only thing that worked! TYSVM! I do happen to be on an M2 but terminal is x86 already.

**Comment by luizbinario** [comment]
Hello. Using POP OS here. I was using node 16 and having the same problem. Upgraded to newer versions to test, but no success. Solved downgrading to 14.21.3.
